sdk/inc/azure/iot/az_iot_hub_client_properties.h

```diff
@@ -0,0 +1,310 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ *
+ * @brief Definition for the IoT Plug and Play properties writer and parsing routines.
+ */
+
+#ifndef _az_IOT_HUB_CLIENT_PROPERTIES_H
+#define _az_IOT_HUB_CLIENT_PROPERTIES_H
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include <azure/core/az_json.h>
+#include <azure/core/az_result.h>
+#include <azure/core/az_span.h>
+
+#include <azure/iot/az_iot_hub_client.h>
+
+#include <azure/core/_az_cfg_prefix.h>
+
+/**
+ * @brief Append the necessary characters to a reported properties JSON payload belonging to a
+ * component.
+ *
+ * The payload will be of the form:
+ *
+ * @code
+ * "reported": {
+ *     "<component_name>": {
+ *         "__t": "c",
+ *         "temperature": 23
+ *     }
+ * }
+ * @endcode
```

Comment:
**Feedback for a separate PR, rather than part of the merge PR:**

I wonder if this renders correctly in doxygen as part of the "brief" which is typically meant to be a sentence.

Maybe this should be a remark or note instead.

Reply:
Agree.  #1893 assigned to me tracks.